spec/features/brexit_checker/accounts_spec.rb

```diff
@@ -3,262 +3,258 @@
+              stub = stub_account_api_set_email_subscription(slug: "your-get-ready-for-brexit-results-a1a2a3a4a5")
+              stub_account_api_set_attributes
               given_i_am_on_the_save_results_confirm_page_with(new_criteria_keys)
               click_on I18n.t("brexit_checker.confirm_changes.save_button")
-              expect(page).to have_content(I18n.t("brexit_checker.confirm_changes_email_signup.heading"))
-            end
-
-            context "the user wants email alerts" do
-              it "creates an email alert" do
-                stub = stub_account_api_set_email_subscription(slug: "your-get-ready-for-brexit-results-a1a2a3a4a5")
-                stub_account_api_set_attributes
-                given_i_am_on_the_save_results_confirm_page_with(new_criteria_keys)
-                click_on I18n.t("brexit_checker.confirm_changes.save_button")
-                find_field(I18n.t("brexit_checker.confirm_changes_email_signup.radio.yes")).click
-                click_on I18n.t("brexit_checker.confirm_changes_email_signup.save_button")
-                expect(stub).to have_been_made
-              end
+              find_field(I18n.t("brexit_checker.confirm_changes_email_signup.radio.yes")).click
+              click_on I18n.t("brexit_checker.confirm_changes_email_signup.save_button")
+              expect(stub).to have_been_made
             end
           end
         end
+      end
 
-        context "the querystring matches what's stored in the account" do
-          it "redirects back to the results page" do
-            given_i_am_on_the_save_results_confirm_page_with(criteria_keys)
-            expect(page).to have_current_path(transition_checker_results_path(c: criteria_keys))
-          end
+      context "the querystring matches what's stored in the account" do
+        it "redirects back to the results page" do
+          given_i_am_on_the_save_results_confirm_page_with(criteria_keys)
+          expect(page).to have_current_path(transition_checker_results_path(c: criteria_keys))
         end
       end
+    end
 
-      def log_in
-        stub_account_api_validates_auth_response(
-          govuk_account_session: Base64.urlsafe_encode64("access-token") + "." + Base64.urlsafe_encode64("refresh-token"),
-        )
-
-        visit transition_checker_new_session_callback_path(state: "state", code: "code")
+    def log_in
+      faked_cookie = Base64.urlsafe_encode64("access-token") + "." + Base64.urlsafe_encode64("refresh-token")
+      headers = { "GOVUK-Account-Session" => faked_cookie }
 
-        @original_account_session_header = page.response_headers["GOVUK-Account-Session"]
-        expect(@original_account_session_header).to_not be_nil
-      end
+      stub_sign_in_callback_url(headers, state: "state", code: "code")
 
-      def log_out
-        visit transition_checker_end_session_path(done: "1")
-        expect(page.response_headers["GOVUK-Account-End-Session"]).to_not be_nil
-      end
```

Comment:
So, it's a bit hard to read with all the diff going on, but I don't think we need these `log_in` and `log_out` methods any more (and so we can remove two of the stubs).  I think the `before` and `after` blocks which call them can be:

```ruby
    before do
      @original_headers = page.driver.options[:headers]
      @original_account_session_header = { "GOVUK-Account-Session" => "placeholder-session" }
      page.driver.options[:headers] ||= {}
      page.driver.options[:headers].merge!("GOVUK-Account-Session" => @original_account_session_header)
    end

    after do
      page.driver.options[:headers] = @original_headers
    end
```